src/main.ts

```diff
@@ -13,6 +15,82 @@ async function setup(): Promise<void> {
   await setOutputs(name, url)
 }
 
+async function promote(): Promise<void> {
+  const user: string = core.getInput('user')
+  const password: string = core.getInput('password')
+  const url: string = core.getInput('url')
+  const name: string = core.getInput('name')
+  const tag: string = core.getInput('tag')
+
+  const staging_regex = new RegExp(
+    '^[^/]+:(staging|prolonged)-([0-9a-f]+).[^./]+.[0-9]+.[0-9]+$'
+  )
+  const promoted_regex = new RegExp('^[^/]+:promoted-([^/]+)$')
+
+  const staging_match = name.match(staging_regex)
+  const promoted_match = name.match(promoted_regex)
+
+  let artifacts_commit = ''
+  if (staging_match !== null) {
+    artifacts_commit = staging_match[2]
+  } else if (promoted_match !== null) {
+    artifacts_commit = promoted_match[1]
+  }
+
+  let myOutput = ''
+
+  const options: ExecOptions = {}
+  options.listeners = {
+    stdout: (data: Buffer) => {
+      myOutput += data.toString()
+    }
+  }
+
+  await exec.exec('git', ['rev-list', '-n', '1', tag], options)
+  if (!myOutput.includes(artifacts_commit))
```

Comment:
I am afraid that if the source build does not match the staging regexp, artifacts_commit will be '' and will be included in the rev-list output. See above for the throw Error('') comment.

Reply:
Well eve work exactly the same way :thinking: 
I think it have never been used to promote already promoted builds

Reply:
It even don't make much sens